lib/ansible/module_utils/remote_management/dellemc/dellemc_ome.py

```diff
@@ -0,0 +1,172 @@
+# -*- coding: utf-8 -*-
+
+#
+# Dell EMC OpenManage Ansible Modules
+# Version 1.3
+# Copyright (C) 2019 Dell Inc.
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# All rights reserved. Dell, EMC, and other trademarks are trademarks of Dell Inc. or its subsidiaries.
+# Other trademarks may be trademarks of their respective owners.
+#
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+import json
+from ansible.module_utils.urls import open_url, ConnectionError, SSLValidationError
+from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
+from ansible.module_utils.six.moves.urllib.parse import urlencode
+
+SESSION_RESOURCE_COLLECTION = {
+    "SESSION": "SessionService/Sessions",
+    "SESSION_ID": "SessionService/Sessions('{Id}')",
+}
+
+
+class OpenURLResponse(object):
+    """Handles HTTPResponse"""
+
+    def __init__(self, resp):
+        self.body = None
+        self.resp = resp
+        if self.resp:
+            self.body = self.resp.read()
+
+    @property
+    def json_data(self):
+        try:
+            return json.loads(self.body)
+        except ValueError:
+            raise ValueError("Unable to parse json")
+
+    @property
+    def status_code(self):
+        return self.resp.getcode()
+
+    @property
+    def success(self):
+        return self.status_code in (200, 201, 202, 204)
+
+    @property
+    def token_header(self):
+        return self.resp.headers.get('X-Auth-Token')
+
+
+class RestOME(object):
+    """Handles OME API requests"""
+
+    def __init__(self, module_params=None, req_session=False):
+        self.module_params = module_params
+        self.hostname = self.module_params["hostname"]
+        self.username = self.module_params["username"]
+        self.password = self.module_params["password"]
+        self.port = self.module_params["port"]
+        self.req_session = req_session
+        self.session_id = None
+        self.protocol = 'https'
+        self._headers = {'Content-Type': 'application/json', 'Accept': 'application/json'}
+
+    def _get_base_url(self):
+        """builds base url"""
+        return '{0}://{1}:{2}/api'.format(self.protocol, self.hostname, self.port)
+
+    def _build_url(self, path, query_param=None):
+        """builds complete url"""
+        url = path
+        base_uri = self._get_base_url()
+        if path:
+            url = '{0}/{1}'.format(base_uri, path)
+        if query_param:
+            url += "?" + urlencode(query_param)
```

Comment:
isn't better to follow the format string style?

Reply:
Updated the code